packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -181,125 +211,560 @@ class CupertinoAlertDialog extends StatelessWidget {
   ///    section when it is long.
   final ScrollController actionScrollController;
 
-  @override
-  Widget build(BuildContext context) {
+  Widget _buildContent() {
     final List<Widget> children = <Widget>[];
 
     if (title != null || content != null) {
-      final Widget titleSection = new _CupertinoAlertTitleSection(
+      final Widget titleSection = new _CupertinoAlertContentSection(
         title: title,
         content: content,
         scrollController: scrollController,
       );
       children.add(new Flexible(flex: 3, child: titleSection));
-      // Add padding between the sections.
-      children.add(const Padding(padding: EdgeInsets.only(top: 8.0)));
     }
 
+    return new Container(
+      color: _kDialogColor,
+      child: new Column(
+        mainAxisSize: MainAxisSize.min,
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        children: children,
+      ),
+    );
+  }
+
+  Widget _buildActions() {
+    Widget actionSection = new Container(
+      height: 0.0,
+    );
     if (actions.isNotEmpty) {
-      final Widget actionSection = new _CupertinoAlertActionSection(
+      actionSection = new _CupertinoAlertActionSection(
         children: actions,
         scrollController: actionScrollController,
       );
-      children.add(
-        new Flexible(child: actionSection),
-      );
     }
 
-    return new Padding(
-      padding: const EdgeInsets.symmetric(vertical: _kEdgePadding),
-      child: new CupertinoDialog(
-        child: new Column(
-          mainAxisSize: MainAxisSize.min,
-          crossAxisAlignment: CrossAxisAlignment.stretch,
-          children: children,
+    return actionSection;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final double textScaleFactor = MediaQuery.of(context).textScaleFactor;
+    return new _CupertinoAccessibility(
+      textScaleFactor: textScaleFactor,
+      child: new MediaQuery(
+        data: MediaQuery.of(context).copyWith(
+          // iOS does not shrink dialog content below a 1.0 scale factor
+          textScaleFactor: math.max(textScaleFactor, 1.0),
+        ),
+        child: new LayoutBuilder(
+          builder: (BuildContext context, BoxConstraints constraints) {
+            return new Center(
+              child: new Container(
+                margin: const EdgeInsets.symmetric(vertical: _kEdgePadding),
+                width: _CupertinoAccessibility.of(context)?.isInAccessibilityMode ?? false
+                    ? _kAccessibilityCupertinoDialogWidth
+                    : _kCupertinoDialogWidth,
+                // The following clip is critical. The BackdropFilter needs to have
+                // rounded corners, but Skia cannot internally create a blurred rounded
+                // rect. Therefore, we have no choice but to clip, ourselves.
```

Comment:
please file a bug (on skia) asking for skia to support blurring only within a path, and then link to it here with a TODO

Reply:
Done.